clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1287,6 +1287,16 @@ public void testCommitAsyncNegativeOffset() {
         assertTrue(mockOffsetCommitCallback.exception instanceof IllegalArgumentException);
     }
 
+    @Test(expected = CommitFailedException.class)
+    public void testCommitOffsetSyncCallbackWithoutNonRetriableException() {
```

Comment:
You may need to change `MockClient.prepareResponse` so that you can pass a requestDurationMs, which you can store in FutureResponse. In `send`, you can compute `ClientResponse.receiveTimeMs` to include `requestDurationMs`. In `poll`, you only return responses where `now + timeoutMs => receiveTimeMs`. Does this make sense?